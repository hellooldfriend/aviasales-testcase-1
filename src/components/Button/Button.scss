@import '../../variables.scss';

.button {
    background: $background_secondary;
    border: none;
    outline: none;
    box-sizing: border-box;
    padding: 15px;
    border-radius: $border_radius_secondary;
    text-align: center;
    cursor: pointer;

    &._active {
        background: $background_active;
        color: $color_button_secondary;
    }
}