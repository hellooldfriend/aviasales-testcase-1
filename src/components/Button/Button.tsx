import React from 'react'
import './Button.scss'

import { IButton } from '../../types'


const Button = (props: IButton) => {

    const cls = ['button']
    if(props.active) {
        cls.push('_active')
    }

    return (
        <button
            className={cls.join(' ')}
            disabled={props.disabled}
            onClick={props.onClick}
        >{props.value}</button>
    )
}

export default Button