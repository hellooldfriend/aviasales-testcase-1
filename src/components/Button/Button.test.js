import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

import Button from './Button'


afterEach(() => {
    cleanup()
})


const props = {
    value: 'button name is here'
}

describe('Button', () => {
    test('Should render not active button', () => {
        render(<Button value={props.value} />)
        const element = screen.getByTestId('button')
        expect(element).toBeInTheDocument()
        expect(element).toHaveTextContent(props.value)
        expect(element).not.toHaveClass('_active')
    })

    test('Should render active button', () => {
        render(
            <Button
                value={props.value}
                active={true}
            />
        )
        const element = screen.getByTestId('button')
        expect(element).toBeInTheDocument()
        expect(element).toHaveTextContent(props.value)
        expect(element).toHaveClass('_active')
    })

    test('Matches snapshot', () => {
        const element = renderer.create(<Button value={props.value} />).toJSON()
        expect(element).toMatchSnapshot()
    })

})

