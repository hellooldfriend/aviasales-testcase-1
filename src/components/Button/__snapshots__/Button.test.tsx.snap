// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Matches snapshot 1`] = `
<button
  className="button"
  data-testid="button"
>
  button name is here
</button>
`;
